### PR TITLE
Feature/add cookies pref

### PR DESCRIPTION
**Pull request by JoshSinghONS**
### What

Updated Regex
Added cookies-pref.js
Updated .yml files to use more recent version of node

### How to review

LGTM

### Who can review

Anyone.

### CODE REVIEW

File: auditjs.json [hunk 1]
```diff
@@ -31,6 +31,10 @@
     {
       "id": "CVE-2020-7608",
       "reason": "yargs-parser@5 is a transitive dependency of gulp-cli which is no longer maintained"
+    },
+    {
+      "id":  "CVE-2022-38900",
+      "reason":"decode-uri-component ... something something ... not sure what to put here."
```
Comment (mr-nick17) [inline auditjs.json:37]: The reason why you are adding it to the exclude list needs to be put here see inspiration above

File: src/js/dp/cookies-pref.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+document.addEventListener("DOMContentLoaded", function () {
+    if (document.querySelectorAll('.cookies-no-js').length > 0) {
+        document.querySelectorAll('.cookies-no-js').forEach((item) => item.classList.add('hidden'))
+    }
+    if (document.querySelectorAll('.cookies-js').length > 0) {
+        document.querySelectorAll('.cookies-js').forEach((item) => item.classList.remove('hidden'))
+    }
+});
```
Comment (mr-nick17) [inline src/js/dp/cookies-pref.js:8]: Empty EOF line is missing

File: package-lock.json [hunk 3]
```diff
@@ -3908,6 +3898,25 @@
         }
       }
     },
+    "glob-parent": {
+      "version": "3.1.0",
```
Comment (mr-nick17) [inline package-lock.json:3902]: This is changing the version to a lower version that will also cause the audit issues that you're ignoring

**Review by mr-nick17 — COMMENTED** [review]
You need to try and fix the audit problems. Try updating node to v20 (or bumping to at least v18-LTS) and then update the dependencies; this should hopefully resolve the audit problem

**Comment by JoshSinghONS** [comment]
I am at a loss. 

The latest commit allows me to pass the audit on my local machine where I run node 20, I'm not sure what the CI/CD pipeline uses.

From what I can tell when I keep resolutions instead of overrides glob-parent does not change it's version for some of the projects its a dependent for

Using "npm ls glob-parent"

dp-design-system@2.0.0 /Users/joshuasingh/src/github.com/ONSdigital/dp-design-system
├─┬ @babel/cli@7.22.5
│ └─┬ chokidar@3.5.3
│   └── glob-parent@5.1.2
├─┬ babel-eslint@10.1.0
│ └─┬ eslint@8.43.0
│   └── glob-parent@6.0.2
└─┬ gulp@4.0.2
  ├─┬ glob-watcher@5.0.5
  │ └─┬ chokidar@2.1.8
  │   └── glob-parent@3.1.0
  └─┬ vinyl-fs@3.0.3
    └─┬ glob-stream@6.1.0
      └── glob-parent@3.1.0
      
You can see the dependencies gulp 4.0.2 has uses an outdated version of glob-parent, alas gulp 4.0.2 is the latest version. Using the overrides property bumps the lower glob-parents to the latest version and that works locally as mentioned but the CI/CD audit still fails.

Any ideas?

File: ci/build.yml [hunk 4]
```diff
@@ -5,8 +5,8 @@ platform: linux
 image_resource:
   type: docker-image
   source:
-    repository: node
-    tag: 14.15.4
+    repository: onsdigital/dp-concourse-tools-node-go
+    tag: latest
```
Comment (lindenmckenzie) [inline ci/build.yml:9]: Think we should still be using the specific tag here rather than `latest`

**Review by lindenmckenzie — CHANGES_REQUESTED** [review]
See comment

File: ci/build.yml (hunk 4 above)
Comment (lindenmckenzie) [reply ci/build.yml:9]: see https://github.com/ONSdigital/florence/blob/develop/ci/lint.yml

**Review by lindenmckenzie — DISMISSED** [review]

**Pull request merged**